libopenage/input/input_manager.cpp

```diff
@@ -12,57 +12,139 @@ namespace input {
+	      { sdl_key(SDLK_F9), actions::ACTION_MAP.at("QUICK_LOAD") },
+	      { sdl_key(SDLK_SPACE), actions::ACTION_MAP.at("TOGGLE_BLENDING") },
+	      { sdl_key(SDLK_F12), actions::ACTION_MAP.at("TOGGLE_PROFILER") },
+	      { sdl_key(SDLK_m), actions::ACTION_MAP.at("TOGGLE_CONSTRUCT_MODE") },
+	      { sdl_key(SDLK_p), actions::ACTION_MAP.at("TOGGLE_UNIT_DEBUG") },
+	      { sdl_key(SDLK_t), actions::ACTION_MAP.at("TRAIN_OBJECT") },
+	      { sdl_key(SDLK_y), actions::ACTION_MAP.at("ENABLE_BUILDING_PLACEMENT") },
+	      { sdl_key(SDLK_z, KMOD_LCTRL), actions::ACTION_MAP.at("DISABLE_SET_ABILITY") },
+	      { sdl_key(SDLK_x, KMOD_LCTRL), actions::ACTION_MAP.at("SET_ABILITY_MOVE") },
+	      { sdl_key(SDLK_c, KMOD_LCTRL), actions::ACTION_MAP.at("SET_ABILITY_GATHER") },
+	      { sdl_key(SDLK_g), actions::ACTION_MAP.at("SET_ABILITY_GARRISON") },
+	      { sdl_key(SDLK_v), actions::ACTION_MAP.at("SPAWN_VILLAGER") },
+	      { sdl_key(SDLK_DELETE), actions::ACTION_MAP.at("KILL_UNIT") },
+	      { sdl_key(SDLK_q), actions::ACTION_MAP.at("BUILDING_1") },
+	      { sdl_key(SDLK_w), actions::ACTION_MAP.at("BUILDING_2") },
+	      { sdl_key(SDLK_e), actions::ACTION_MAP.at("BUILDING_3") },
+	      { sdl_key(SDLK_r), actions::ACTION_MAP.at("BUILDING_4") },
+	      { sdl_key(SDLK_q, KMOD_LCTRL), actions::ACTION_MAP.at("BUILDING_5") },
+	      { sdl_key(SDLK_w, KMOD_LCTRL), actions::ACTION_MAP.at("BUILDING_6") },
+	      { sdl_key(SDLK_e, KMOD_LCTRL), actions::ACTION_MAP.at("BUILDING_7") },
+	      { sdl_key(SDLK_r, KMOD_LCTRL), actions::ACTION_MAP.at("BUILDING_8") },
+	      { sdl_key(SDLK_z), actions::ACTION_MAP.at("BUILDING_TOWN_CENTER") },
+	      { sdl_key(SDLK_1), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_1") },
+	      { sdl_key(SDLK_2), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_2") },
+	      { sdl_key(SDLK_3), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_3") },
+	      { sdl_key(SDLK_4), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_4") },
+	      { sdl_key(SDLK_5), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_5") },
+	      { sdl_key(SDLK_6), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_6") },
+	      { sdl_key(SDLK_7), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_7") },
+	      { sdl_key(SDLK_8), actions::ACTION_MAP.at("SWITCH_TO_PLAYER_8") },
+	      { sdl_key(SDLK_UP), actions::ACTION_MAP.at("UP_ARROW") },
+	      { sdl_key(SDLK_DOWN), actions::ACTION_MAP.at("DOWN_ARROW") },
+	      { sdl_key(SDLK_LEFT), actions::ACTION_MAP.at("LEFT_ARROW") },
+	      { sdl_key(SDLK_RIGHT), actions::ACTION_MAP.at("RIGHT_ARROW") },
+	      { sdl_mouse(1), actions::ACTION_MAP.at("PAINT_TERRAIN") },
+	      { sdl_mouse(1), actions::ACTION_MAP.at("SELECT") },
+	      { sdl_mouse(3), actions::ACTION_MAP.at("DESELECT") },
+	      { sdl_wheel(1), actions::ACTION_MAP.at("FORWARD") },
+	      { sdl_wheel(-1), actions::ACTION_MAP.at("BACK") }},
 	relative_mode{false} {
 }
 
+std::string InputManager::get_key(const std::string action_str) {
+	for (auto it=keys.begin(); it!=keys.end(); ++it) {
```

Comment:
you can just use `for (auto &it : this->keys) { it.first.stuff; it.second.bla(); }`